container/docker/factory.go

```diff
@@ -139,6 +139,12 @@ func FullContainerName(dockerId string) string {
 func (self *dockerFactory) CanHandleAndAccept(name string) (bool, bool, error) {
 	// docker factory accepts all containers it can handle.
 	canAccept := true
+
+	// When using SystemD all docker containers have a docker- prefix
+	if UseSystemd() && !strings.HasPrefix(path.Base(name), "docker-") {
+		return false, canAccept, nil
```

Comment:
I think this might not be true if a docker container is started with --cgroup_parent flag. parent cgroup name will replace docker in that case.